pkg/rancher-desktop/main/diagnostics/limaDarwin.ts

```diff
@@ -0,0 +1,57 @@
+import semver from 'semver';
+
+import { DiagnosticsCategory, DiagnosticsChecker } from './types';
+
+import mainEvents from '@pkg/main/mainEvents';
+import { spawnFile } from '@pkg/utils/childProcess';
+import Logging from '@pkg/utils/logging';
+
+const console = Logging.diagnostics;
+
+let kubernetesMemory = Number.POSITIVE_INFINITY;
+
+mainEvents.on('settings-update', (cfg) => {
+  kubernetesMemory = cfg.kubernetes.memoryInGB;
+});
+
+/**
+ * CheckLimaDarwin version checks for compatibility with the host machine.
+ */
+const CheckLimaDarwin: DiagnosticsChecker = {
+  id:       'LIMA_DARWIN_VERSION',
+  category: DiagnosticsCategory.ContainerEngine,
+  applicable() {
+    const isDarwin = process.platform === 'darwin';
+    const isArm = process.arch.startsWith('arm');
+
+    return Promise.resolve(isDarwin && isArm);
+  },
+  async check() {
+    const result = {
+      description:   '',
+      passed:        false,
+      fixes:         [] as { description: string }[],
+    };
+    const { stdout } = await spawnFile('/usr/bin/sw_vers', ['-productVersion'], { stdio: ['ignore', 'pipe', console] });
+    let currentVersion = stdout.trim();
+
+    if (!/^\d+\.\d+\.\d+/.test(currentVersion)) {
+      currentVersion += '.0'; // massage it into looking like semver
+    }
```

Comment:
Instead of spawning `sw_vers` and parsing the result, you could use the Darwin version from `os.version()`.

According to https://en.wikipedia.org/wiki/Darwin_%28operating_system%29#Darwin_20_onwards macOS 12.4 corresponds to darwin version 21.5.0.